### PR TITLE
[Vue] Pass DOM Element in the config object instead of editor constructor in CKEditor >= 48

### LINKED ISSUE #400: [Vue] Pass DOM Element in the config object instead of editor constructor in CKEditor >= 48
### Description

Following the core architectural decisions implemented in https://github.com/ckeditor/ckeditor5/issues/19887, the editor's initialization API has changed. 

The DOM element should now be passed through the configuration object (`config.attachTo` for Classic Editor) rather than being passed as the first argument to the editor's constructor or the static `create()` method. The old signatures are now deprecated.

### PR DESCRIPTION
**Pull request by Mati365**
### 🚀 Summary

[Vue] Pass DOM Element in the config object instead of editor constructor in CKEditor >= 48

---

### 📌 Related issues

* Closes https://github.com/ckeditor/ckeditor5-vue/issues/400

### CODE REVIEW

File: tests/_utils/setCKEditorVersion.ts
Comment (gorzelinski) [inline tests/_utils/setCKEditorVersion.ts:1]: Do we need this function? It doesn't seem to be imported anywhere.

File: tests/utils/assignInitialDataToEditorConfig.test.ts
Comment (gorzelinski) [inline tests/utils/assignInitialDataToEditorConfig.test.ts:1]: We still have an old setup in the mock. Shouldn't we update it? (The comment is about the file mentioned below.)
https://github.com/ckeditor/ckeditor5-vue/blob/master/tests/_utils/mockeditor.ts#L58

File: src/ckeditor.vue [hunk 1]
```diff
@@ -150,17 +154,28 @@ function setUpEditorEvents( editor: EditorType ) {
...
+
 	if ( model.value ) {
-		editorConfig = assignInitialDataToEditorConfig( editorConfig, model.value );
+		editorConfig = assignInitialDataToEditorConfig( editorConfig, model.value, true );
 	}
 
-	( props.editor.create( element.value, editorConfig ) as unknown as Promise<EditorType> )
+	const editorPromise = (
+		supports.elementConfigAttachment ?
+			Editor.create( assignElementToEditorConfig( Editor, element.value!, editorConfig ) ) :
```
Comment (gorzelinski) [inline src/ckeditor.vue:174]: Can we be sure about the element value `element.value!`? Won't there be any edge cases?

File: tests/_utils/setCKEditorVersion.ts
Comment (Mati365) [reply tests/_utils/setCKEditorVersion.ts:1]: good catch

File: src/ckeditor.vue (hunk 1 above)
Comment (Mati365) [reply src/ckeditor.vue:174]: Nope, `element.value` is always filled in `onMounted`

File: tests/utils/assignInitialDataToEditorConfig.test.ts
Comment (Mati365) [reply tests/utils/assignInitialDataToEditorConfig.test.ts:1]: Fixed in https://github.com/ckeditor/ckeditor5-vue/pull/403/commits/0df6880c6fda90de215149b8f68a82bf277fe926

**Review by gorzelinski — APPROVED** [review]
